samples/Calling/src/app/App.tsx

```diff
@@ -22,207 +18,135 @@ import {
-    const joiningExistingCall: boolean = page === 'home' && (!!getGroupIdFromUrl() || !!getTeamsLinkFromUrl());
-
-    switch (page) {
-      case 'home': {
-        return (
-          <HomeScreen
-            joiningExistingCall={joiningExistingCall}
-            startCallHandler={(callDetails) => {
-              setDisplayName(callDetails.displayName);
-              const isTeamsCall = !!callDetails.teamsLink;
-              const callLocator =
-                callDetails.teamsLink || getTeamsLinkFromUrl() || getGroupIdFromUrl() || createGroupId();
-              setCallLocator(callLocator);
-              setCallJoinType(joiningExistingCall ? (isTeamsCall ? 'teamsCall' : 'joinCall') : 'newCall');
-
-              // Update window URL to have a joinable link
-              if (!joiningExistingCall) {
-                const joinParam = isTeamsCall
-                  ? '?teamsLink=' + encodeURIComponent((callLocator as TeamsMeetingLinkLocator).meetingLink)
-                  : '?groupId=' + (callLocator as GroupCallLocator).groupId;
-                window.history.pushState({}, 'Joining Call', window.location.origin + joinParam);
-              }
-
-              setPage('configuration');
-            }}
-          />
-        );
-      }
-      case 'endCall': {
-        return <EndCall rejoinHandler={() => setPage('createCallClient')} homeHandler={navigateToHomePage} />;
-      }
-      case 'callError': {
-        return <CallError rejoinHandler={() => setPage('createCallClient')} homeHandler={navigateToHomePage} />;
-      }
-      case 'teamsMeetingDenied': {
-        return (
-          <CallError
-            title="Error joining Teams Meeting"
-            reason="Access to the Teams meeting was denied."
-            rejoinHandler={() => setPage('createCallClient')}
-            homeHandler={navigateToHomePage}
-          />
-        );
-      }
-      case 'removed': {
+    case 'teamsMeetingDenied': {
+      return (
+        <CallError
+          title="Error joining Teams Meeting"
+          reason="Access to the Teams meeting was denied."
+          rejoinHandler={() => setPage('call')}
+          homeHandler={navigateToHomePage}
+        />
+      );
+    }
+    case 'removed': {
+      return (
+        <CallError
+          title="Oops! You are no longer a participant of the call."
+          reason="Access to the meeting has been stopped"
+          rejoinHandler={() => setPage('call')}
+          homeHandler={navigateToHomePage}
+        />
+      );
+    }
+    case 'call': {
+      if (userCredentialFetchError) {
         return (
           <CallError
-            title="Oops! You are no longer a participant of the call."
-            reason="Access to the meeting has been stopped"
-            rejoinHandler={() => setPage('createCallClient')}
+            title="Error getting user credentials from server"
+            reason="Ensure the sample server is running."
```

Comment:
Does this also handle if you start up the server but you forgot to set your key?

Reply:
Yeah - the client error message is always generic but the server will output a good error though when the token is requested